sdk/servicebus/Azure.Messaging.ServiceBus/assets/functions/azure-resource-functions.ps1

```diff
@@ -0,0 +1,427 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
I think we shouldn't add more ways to setup test resources and follow the common pattern setup by @heaths and @danieljurek.

Reply:
This script is intended to bootstrap the environment for contributors, and should offer a reasonable starting point to get the SB infrastructure going.    Happy to revisit and see if I can better plug into what Heath and Daniel did at a later point, but I'm going to leave this for now due to time constraints and immediate need.

Reply:
Let’s take the time to do it the same way as the other services. If we introduce a new way now it will be difficult to change later. The pattern is not just for external contributors but also for our own efficiency.

What would it take to follow the existing pattern?

Reply:
> What would it take to follow the existing pattern?

Delaying other deliverables for this week to dig into and understand the new pattern and absorbing additional efforts to update any live tests that the team is writing between now and when this PR closes.    

Not sure that I agree with the statement that it will be difficult to change later; the asset in question is a script run manually by a contributor - internal or external - to bootstrap their environment for the first time.  It does not play a role in runs, either locally or nightly.   I understand and support the desire to make things uniform, but I'm not sure that I understand the concern behind a short delay in doing so.

Reply:
@jsquire and I spoke about this offline. To get unblocked we'll be using a `test-resources.json` file which will handle dynamic resource creation for the live test pipelines like our other services. 

Any provisioning script that contributors use should call into the `New-TestResources.ps1` script when setting up the user's environment. This will create a resource group and deploy resources specified in the `test-resources.json` file and potentially save duplicated work there.

Reply:
Thanks to @danieljurek for taking the time to talk things over.   I've got an understanding of what needs to be done.   For now, I've pulled the bootstrap script and contributing guide;  I'll convert it over in the near future and introduce the engineering system tie-in as a separate PR.